src/Parsers.jl

```diff
@@ -205,7 +205,7 @@ function parse_status_line!(bytes::AbstractString, response)::SubString{String}
     if !exec(re, bytes)
         throw(ParseError(:INVALID_STATUS_LINE, bytes))
     end
-    response.version = VersionNumber(group(1, re, bytes))
+    response.version = HTTPVersion(group(1, re, bytes))
```

Comment:
here, and a few lines above, is where we were allocating as part of construction `VersionNumber`